### ISSUE TITLE
Does pipline support import 3rd party libraries?

### ISSUE THREAD

**Issue by LipingMao** [ciscopipeline/Documentation#2]
One quick question, if I want to call 3rd party libraries, for example: 
http://pyral.readthedocs.io/en/latest/overview.html

How can I do this in pipline? Thanks.

**Comment by vallard** [comment]
Right now, unfortunately, we don't have this ability.  We are, however making it possible to do in the future.  We will add the ability to add files that can be added by pip.  In the backend we create a new container for each bot.  The bot backend you create you'll just need to add the additional libraries to be installed by pip.  
I'm curious to know more about your usecase with using Rally?

**Comment by LipingMao** [comment]
Thanks @vallard for your quick confirm. I think it would be useful if we can install some 3rd party lib in the container.

My use case is I want to write a spark bot to do some integration with Rally, since rally already have python library, so I'd like to use the library directly.  Another case for example,  I may want to call openstack api with novaclient/neutronclient, but since I can't install these client in pipeline, I have to call restful api directly in the code right now.

**Comment by vallard** [comment]
Yes, I definitely need to do that.  I will be adding a part to pipeline where you can add libraries to the python stuff.  Since I just create a container underneath then not a problem to add.  Will take a few weeks.

**Comment by vallard** [comment]
Sorry I haven't gotten on this but still looking to add it in.
